lib/Parse/ASTGen.cpp

```diff
@@ -100,25 +519,31 @@ StringRef ASTGen::copyAndStripUnderscores(StringRef Orig, ASTContext &Context) {
   return StringRef(start, p - start);
 }
 
-StringRef ASTGen::copyAndStripUnderscores(StringRef Orig) {
-  return copyAndStripUnderscores(Orig, Context);
+SourceLoc ASTGen::advanceLocBegin(const SourceLoc &Loc, const Syntax &Node) {
+  return Loc.getAdvancedLoc(Node.getAbsolutePosition().getOffset());
```

Comment:
This is a bit fragile. This works only if the node we are dealing with is the root node in the syntax tree.

Reply:
I disagree.
Here `Loc` always points to the start of leading trivia of the tree where `Node` is.
`Node.getAbsolutePosition()` traverses from `Node` up to the root node and accumulates length of the visited nodes. When you add this length (offset of `Node` in its tree) to source location of the beginning of the tree (its root), you should get precisely the `SourceLoc` of `Node` I think.

Reply:
> `Loc` always points to the start of leading trivia of the tree where `Node` is.

Ah, I see. Thanks for explanation 👍